cmd/kubeadm/app/apis/kubeadm/validation/validation.go

```diff
@@ -381,13 +381,25 @@ func ValidateIPNetFromString(subnet string, minAddrs int64, fldPath *field.Path)
 	return allErrs
 }
 
+// ValidatePodSubnetsFromString validates network portion of pod subnet(s)
+func ValidatePodSubnetsFromString(cidrStr string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	cidrs := strings.Split(cidrStr, ",")
+	for _, cidr := range cidrs {
+		if _, _, err := net.ParseCIDR(cidr); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, cidr, "couldn't parse subnet"))
+		}
+	}
+	return allErrs
+}
+
 // ValidateNetworking validates networking configuration
 func ValidateNetworking(c *kubeadm.Networking, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, apivalidation.ValidateDNS1123Subdomain(c.DNSDomain, field.NewPath("dnsDomain"))...)
-	allErrs = append(allErrs, ValidateIPNetFromString(c.ServiceSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("serviceSubnet"))...)
+	allErrs = append(allErrs, ValidateServiceSubnetFromString(c.ServiceSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("serviceSubnet"))...)
 	if len(c.PodSubnet) != 0 {
-		allErrs = append(allErrs, ValidateIPNetFromString(c.PodSubnet, constants.MinimumAddressesInServiceSubnet, field.NewPath("podSubnet"))...)
+		allErrs = append(allErrs, ValidatePodSubnetsFromString(c.PodSubnet, field.NewPath("podSubnet"))...)
```

Comment:
this is only going to be a comma separated list if the feature is enable. right?
` if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) ` , otherwise it has to work as always

Reply:
so on slack i said "maybe we can avoid feature gates for this", but we probably shouldn't given we are toggling an alpha feature for a number of components that can cause problems that we cannot predict:
https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go#L538

kubeadm maintains it's own way of doing feature gates (for reasons).
please, have a look at:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/features/features.go#L41
the unit tests should provide an example of how to use this?

kubeadm feature gates are supported in the ClusterConfiguration object (as long as they are registered), they are also supported on the CLI side but we need to double check if all `init` phases have the `--feature-gates` flag propagated.

we also need to organize the tasks better:
- first PR that we merge should add the kubeadm gate (should have a release note about the gate addition)
- second PR (this one) should modify the validation to support ",". examine if validation changes are needed based on the gate.
- third PR should add e2e tests under "kubeadm-e2e"

more comments on the implementation:
- as mentioned above validators should/might behave differently, but in my thinking they should not:
  - passing a comma separated list from the config or CLI should be tolerated by the new validators
  - a comma separated list is backwards compatible and a super-set of field that has a single value, therefore the new validators should can a superset of the of previous ones.
  - this lets us get away with supporting "," separated lists in the old configs - e.g. v1beta2
- if the kubeadm gate is enabled all components such as kubelet, control-plane, need the respected k8s feature gate to be passed to them.
- we need to maintain the kubeadm gate until the feature is GA. once it's GA we have to deprecate the feature gate.
- if we decide to deprecate the feature gate we need to follow the deprecation policy of the "state" it's currently at e.g. longer for GA.

@Arvinderpal please ping me on the PR that adds the kubeadm feature gate.
thanks!

Reply:
> so on slack i said "maybe we can avoid feature gates for this", but we probably shouldn't given we are toggling an alpha feature for a number of components that can cause problems that we cannot predict:

Ok. I will look at adding feature-gate in kubeadm as well. 

> we also need to organize the tasks better:
> 
> * first PR that we merge should add the kubeadm gate (should have a release note about the gate addition)
> * second PR (this one) should modify the validation to support ",". examine if validation changes are needed based on the gate.
> * third PR should add e2e tests under "kubeadm-e2e"

Ok. I have updated the tracking ticket and will use that to track current and future tasks. https://github.com/kubernetes/kubeadm/issues/1612

 
> @Arvinderpal please ping me on the PR that adds the kubeadm feature gate.

Will do! Thanks.